File: All-Topics/2951-3000/3000-MaximumAreaofLongestDiagonalRectangle.java


class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {

      int rows=dimensions.length;
      double result=0;
      int resultArea=0;
      for(int i=0;i<rows;i++){
        int length=dimensions[i][0];
        int width=dimensions[i][1];
        int area=length*width;
        double diagonal=Math.pow(((length*length)+(width*width)),0.5);
        if(diagonal>result){
          result=diagonal;
          resultArea=area;
        }else if(diagonal==result && area>resultArea){
          result=diagonal;
          resultArea=area;
        }
      }

return resultArea;

    }
}

File: All-Topics/2951-3000/3000-MaximumAreaofLongestDiagonalRectangle.ts
const dimensions = [
  [9, 3],
  [8, 6],
]

/**
 * @param {number[][]} dimensions
 * @return {number}
 */
var areaOfMaxDiagonal = function (dimensions: number[][]) {
  let maxArea: number[] = []

  dimensions.map((val) => {
    const diagonal = (val[0] ** 2 + val[1] ** 2) ** 0.5

    const area = val[0] * val[1]
    if (diagonal > (maxArea?.[1] ?? 0)) {
      maxArea = [area, diagonal]
    } else if (diagonal === maxArea?.[1] && (maxArea?.[0] ?? 0) < area) {
      maxArea = [area, diagonal]
    }
  })

  return maxArea[0]
}

console.log(areaOfMaxDiagonal(dimensions))
